src/content/docs/en/guides/internationalization.mdx

```diff
@@ -0,0 +1,325 @@
+
+Your `/[lang]/` folder names must match exactly the `locales` in the list, and every supported language must have its own localized folder. 
+
+Depending on your deploy host, you may discover transformations in URL paths, so check your deployed site to determine the best syntax for your project.
+
+### Routing strategy 
+
+You can choose whether or not pages in your `defaultLocale` should be prefixed with their `/lang/` in their URL path. Every other supported locale will automatically be prefixed.
+
+#### `'prefix-other-locales'`
+
+```js title="astro.config.mjs" ins={7}
+import { defineConfig } from "astro/config"
+export default defineConfig({
+  experimental: {
+    i18n: {
+      defaultLocale: "en",
+      locales: ["es", "en", "fr"],
+      routingStrategy: "prefix-other-locales"
+    }
+  }
+})
+```
+
+This is the **default** value. With this option, URLs in your default language will **not** have a `/[lang]/` prefix.
+
+For example, if `i18n.defaultLocale: "en"` is configured:
+
+- `src/pages/en/blog.astro` will produce the route `example.com/blog/` 
+- `src/pages/fr/blog.astro` will produce the route `example.com/fr/blog/` 
+- If there is no file at `src/pages/es/blog.astro`, then the route `example.com/es/blog/` will 404 unless you specify a [fallback strategy](#fallback).
+
+
+#### `'prefix-always'`
+
+```js title="astro.config.mjs" ins={7}
+import { defineConfig } from "astro/config"
+export default defineConfig({
+  experimental: {
+    i18n: {
+      defaultLocale: "en",
+      locales: ["es", "en", "fr"],
+      routingStrategy: "prefix-always"
+    }
+  }
+})
+```
+
+With this option, all routes will have their locale code (prefix) in their URL.
+
+- `src/pages/[lang]/blog.astro` will all reproduce routes of the form `example.com/[locale]/blog/`.
+- The URL `example.com/blog/` will return a 404 (not found) status code.
+
+
+### Browser language detection
+
+Astro's i18n routing combined with one of Astro's  [on-demand server rendering modes](/en/guides/server-side-rendering/) (`output:'server'` or `output:'hybrid'`) allow you to access two properties for browser language detection: `Astro.preferredLocale` and `Astro.preferredLocaleList`.
+
+These combine the browser's `Accept-Langauge` header, and your `locales` to automatically respect your visitor's preferred languages.
+
+- `Astro.preferredLocale`:  Astro can compute a **preferred locale** for your visitor if their browser's preferred locale is included in your `locales` array. This value is undefined if no such match exists.
+
+
+- `Astro.preferredLocaleList`: An array of all locales that are both requested by the browser and supported by your website. This produces a list of all compatible languages between your site and your visitor. The value is `[]` if none of the browser's requested languages are found in your `locales` array. If the browser does not specify any preferred languages, then this value will be [`i18n.locales`].
```

Comment:
I am not sure if it's worth mentioning it, but I leave the information here, and you decide: the locales coming from `Accept-Language` have their format, e.g. `pt-BR`, although it's possible that developers decide to use a different format inside `i18n.locales`, e.g. `pt_BR`, `pt-br`. 

Astro under to hood does some transformations (no need to mention this) and eventually will return the format **specified in `i18n.locales`**. 

So of `Accept-Language` is `fr-CA;q=0.1, pt-BR;q=0.5", and `i18n.locales` is `["fr_CA", "pt_BR"]`:
- `Astro.preferredLocale` is `"pt_BR"`;
- `Astro.preferredLocaleList` is `["pt_BR", "fr_CA"]`, the order is important because it's kept from `Accept-Language`

Reply:
Let me see what I can do here!